drivers/storage/portworx/component/pvccontroller.go

```diff
@@ -551,6 +568,31 @@ func (c *pvcController) getPVCControllerDeploymentSpec(
 	return deployment
 }
 
+func (c *pvcController) isK3sDeployment() bool {
+
+	if c.isK3s != nil {
+		return *c.isK3s
+	} else {
+		// Fetch the list of nodes
+		nodes, err := coreops.Instance().GetNodes()
+		if err != nil {
+			logrus.Errorf("error during fetching nodes %s", err)
+			return false
+		}
+
+		// Iterate over nodes and check their versions for "k3s"
+		for _, node := range nodes.Items {
+			if strings.Contains(node.Status.NodeInfo.KubeletVersion, "k3s") {
+				c.isK3s = boolPtr(true)
+				return true
+			}
+		}
+		c.isK3s = boolPtr(false)
```

Comment:
This is not incorrect, but we already have a way in the operator code to detect a k3s cluster from the k8s server version string. https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/deployment.go#L1951

Let's keep it consistent and use the same. I would recommend modifying the `isK3sClusterExt` method and extracting out the `strings.HasPrefix(ext[1:], "k3s")` part into `portworx/util.go`. Then call the same from here.